_episodes/08-putting-it-all-together.md

```diff
@@ -86,209 +89,216 @@ also change the name of the columns in the DataFrame like this:
-> values. Add axis labels, titles, and legends to your figures. Make at least one
-> figure with multiple plots using the function `subplot()`.
-{: .challenge}
+## Matplotlib package
 
-### Using pyplot:
+[Matplotlib](https://matplotlib.org/) is a Python package that is widely used throughout the scientific Python community to create high-quality and publication-ready graphics. It supports a wide range of raster and vector graphics formats including PNG, PostScript, EPS, PDF and SVG.
 
-First, import the pyplot toolbox:
+Moreover, matplotlib is the actual engine behind the plotting capabilities of both the Pandas and the plotnine package. For example, when we call the `.plot` method on Pandas data objects, we actually use the matplotlib package:
 
 ~~~
-    import matplotlib.pyplot as plt
+import pandas as pd
+surveys = pd.read_csv("data/surveys.csv")
+my_plot = surveys.plot("hindfoot_length", "weight", kind="scatter")
+my_plot
 ~~~
 {: .language-python}
 
-By default, matplotlib will create the figure in a separate window. When using
-ipython notebooks, we can make figures appear in-line within the notebook by
-writing:
+![Scatter plot of survey data set](../fig/08_scatter_surveys.png)
+
+The returned object is a matplotlib object (check it yourself with `type(my_plot)`) and the power of matplotlib is available to further adjust these plots as it is created with matplotlib itself.
+
+> ## Tip
+> Matplotlib itself can be overwhelming, so a useful strategy is to 
+> do as much as you easily can in a convenience layer, i.e. start
+> creating the plot in Pandas or plotnine, and then use matplotlib
+> for the rest.
+{: .callout}
+
+We will cover a few basic commands for creating and formatting plots with matplotlib in this lesson. A great resource for help creating and styling your figures is the matplotlib gallery
+(<http://matplotlib.org/gallery.html>), which includes plots in many different
+styles and the source code that creates them. 
+
+### Using the matplotlib package
+
+First, import the pyplot toolbox:
 
 ~~~
-    %matplotlib inline
+import matplotlib.pyplot as plt
 ~~~
 {: .language-python}
 
-We can start by plotting the values of a list of numbers (matplotlib can handle
-many types of numeric data, including numpy arrays and pandas DataFrames - we
-are just using a list as an example!):
+By default, matplotlib creates a figure in a separate window. When using
+Jupyter notebooks, we can make figures appear in-line within the notebook by
+writing:
 
 ~~~
-    list_numbers = [1.5, 4, 2.2, 5.7]
-    plt.plot(list_numbers)
-    plt.show()
+%matplotlib inline
 ~~~
 {: .language-python}
 
-The command `plt.show()` prompts Python to display the figure. Without it, it
-creates an object in memory but doesn't produce a visible plot. The ipython
-notebooks (if using `%matplotlib inline`) will automatically show you the figure
-even if you don't write `plt.show()`, but get in the habit of including this
-command!
```

Comment:
`.show()` was previously mentioned and was, in fact, recommended. I'm not a big Jupyter user (`vi` habits take precedence), so I'd argue that it has to be at least mentioned.

Reply:
Coming back from maternity leave this week, so I'm getting back at looking at the repo. Is the change requested here to make clear that .show() should be included for portability? If this is the last thing standing between merging and not, I can make this change. I just want to make sure I understand the objection.

Reply:
those who don't use Jupyter notebooks will not see anything without the `.show()` command. While I appreciate all the benefit that Jupyter Notebook brings, I feel it is important to emphasize the fact that it does some things behind the scenes that learners will have to remember to do themselves in any other environment.

Reply:
OK, I see the issue. I can fix that (as an also Vim-defaulter). I can merge this (since it fixes several things in our issue queue) and add the language in a separate commit. Does that sound all right?

Reply:
You can add changes directly to this branch. I'll send you a message in Slack.

Reply:
Good point @maxim-belkin I would indeed keep this text there to make sure non-notebook users are having the required info.